Sources/Client.swift

```diff
@@ -173,18 +183,66 @@ open class Client {
 
 
 extension Client {
+    /**
+     Fetch a collection of Entries from Contentful matching the specified query. This method does not
+     specify the content_type in the query parameters, so the entries returned in the results can be
+     of any type.
 
-    @discardableResult public func fetchEntries<ContentType: EntryModel>
-        (with query: Query<ContentType>, completion: @escaping (Result<[ContentType]>) -> Void) -> URLSessionDataTask? {
+     - Parameter query: The Query object to match results againts.
+     - Parameter completion: A handler being called on completion of the request.
 
-        let url = URL(forComponent: "entries", parameters: query.queryParameters())
+     - Returns: The data task being used, enables cancellation of requests.
+     */
+    @discardableResult public func fetchEntries(with query: Query, completion: @escaping ResultsHandler<[Entry]>) -> URLSessionDataTask? {
```

Comment:
About `@discardableResult` does this mean you can call `fetchEntries` without caring for its return value? I am failing to understand the usefullness ...